astropy/table/index.py

```diff
@@ -829,18 +834,24 @@ def __init__(self, table):
         if len(self.indices) == 0:
             raise ValueError("Cannot create TableLoc object with no indices")
 
-    def _get_rows(self, item):
+    def _get_rows(
+        self,
+        item: int | Iterable[int] | slice | tuple[str, slice],
+    ) -> list[np.int64]:
```

Comment:
obviously, these annotations are not part of the fix, rather a byproduct of my time spent looking for the bug :)